modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -1384,27 +1388,44 @@ private[smithy4s] class SchemaVisitorJCodec(
       )
   }
 
+  private def writeLabel(label: String, out: JsonWriter): Unit =
+    if (label.forall(JsonWriter.isNonEscapedAscii)) {
+      out.writeNonEscapedAsciiKey(label)
+    } else out.writeKey(label)
+
   private def fieldEncoder[Z, A](
       field: Field[Z, A]
   ): (Z, JsonWriter) => Unit = {
     val codec = apply(field.schema)
     val jLabel = jsonLabel(field)
-    val writeLabel: JsonWriter => Unit =
-      if (jLabel.forall(JsonWriter.isNonEscapedAscii)) {
-        _.writeNonEscapedAsciiKey(jLabel)
-      } else _.writeKey(jLabel)
-
     if (explicitDefaultsEncoding) { (z: Z, out: JsonWriter) =>
-      writeLabel(out)
+      writeLabel(jLabel, out)
       codec.encodeValue(field.get(z), out)
     } else { (z: Z, out: JsonWriter) =>
       field.foreachUnlessDefault(z) { (a: A) =>
-        writeLabel(out)
+        writeLabel(jLabel, out)
         codec.encodeValue(a, out)
       }
     }
   }
 
+  private def jsonUnknownFieldEncoder[Z, A](
+      field: Field[Z, A]
+  ): (Z, JsonWriter) => Unit = { (z: Z, out: JsonWriter) =>
+    field.foreachUnlessDefault(z) {
+      case m: Map[_, _] =>
+        m.foreach { case (label: String, value: Document) =>
+          writeLabel(label, out)
+          documentJCodec.encodeValue(value, out)
+        }
+      case Some(m: Map[_, _]) =>
+        m.foreach { case (label: String, value: Document) =>
+          writeLabel(label, out)
+          documentJCodec.encodeValue(value, out)
+        }
```

Comment:
Sorry, I should have signposted this bit as well, but this is falling under the similar notion to what I was describing [here](https://github.com/disneystreaming/smithy4s/pull/1574#discussion_r1713392500) : 

1. You cannot make the assumption that a smithy map necessarily translates to a Scala map 
2. You cannot make the assumption that the absence of `required` translates to Scala Option 
3. You cannot even make the assumption that at this level, you're dealing with `Documents`

Therefore you need to pre-compile the `Document.Encoder` and encode the field value against it, and pattern-match against the produced document to check it's a DObject, in which case you run the exact same `foreach` you have there